src/scheduler.h

```diff
@@ -5,36 +5,30 @@
 #ifndef BITCOIN_SCHEDULER_H
 #define BITCOIN_SCHEDULER_H
 
-//
-// NOTE:
-// boost::thread should be ported to std::thread
-// when we support C++11.
-//
 #include <condition_variable>
 #include <functional>
 #include <list>
 #include <map>
 
 #include <sync.h>
 
-//
-// Simple class for background tasks that should be run
-// periodically or once "after a while"
-//
-// Usage:
-//
-// CScheduler* s = new CScheduler();
-// s->scheduleFromNow(doSomething, std::chrono::milliseconds{11}); // Assuming a: void doSomething() { }
-// s->scheduleFromNow([=] { this->func(argument); }, std::chrono::milliseconds{3});
-// boost::thread* t = new boost::thread(std::bind(CScheduler::serviceQueue, s));
-//
-// ... then at program shutdown, make sure to call stop() to clean up the thread(s) running serviceQueue:
-// s->stop();
-// t->join();
-// delete t;
-// delete s; // Must be done after thread is interrupted/joined.
-//
-
+/**
+ * Simple class for background tasks that should be run
+ * periodically or once "after a while"
+ *
+ * Usage:
+ *
+ * CScheduler* s = new CScheduler();
```

Comment:
Not sure how to preserve this part of the comment then:

```
delete s; // Must be done after thread is interrupted/joined.
```

If this is changed, I'd rather just

```
CScheduler s{};
s.scheduleFromNow(doSomething, std::chrono::milliseconds{11}); // Assuming a: void doSomething() { }
s.scheduleFromNow([=] { this->func(argument); }, std::chrono::milliseconds{3});
std::thread t{[&] { s.serviceQueue(); }};
 
 ... then at program shutdown, make sure to call stop() to clean up the thread(s) running serviceQueue:
s.stop();
t.join();

Reply:
That looks reasonable to me, although I don't think it's _that_ important. If you don't think it matters, leave it as is to avoid bikeshedding.

However, I do think that you should s/stop()/StopImmediately()/ in this comment now that you're removing the `stop()` function.

Reply:
Left as is for now.